pkg/tag/update.go

```diff
@@ -63,3 +70,163 @@ func EnsureAliasesUnique(id int, aliases []string, qb models.TagReader) error {
 
 	return nil
 }
+
+func EnsureUniqueHierarchy(id int, parentIDs, childIDs []int, qb models.TagReader) error {
```

Comment:
This needs unit tests.

Reply:
Added.

I did notice some test files have a "marker" for `// +build integration` while others don't, and these also seem to be in `package X_test`. Not sure if I should have added that as well to the test file (I never wrote any unit tests in Go so not sure what the "marker" + different package really means)

Reply:
`// +build integration` is used where we need to run against a database. Your tests should run without a database, so it should be fine without.

`_test` package is used to enforce internal API access restrictions, and isn't necessary.